lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -5281,13 +5281,27 @@ LogicalResult ConvertAtenOp<AtenSqrtOp>::matchAndRewrite(
     AtenSqrtOp op, OpAdaptor adaptor,
     ConversionPatternRewriter &rewriter) const {
 
-  // Converts AtenSqrtOp into (Reciprocal + Rsqrt)
-  Value self = adaptor.getSelf();
-  auto rcpOp =
-      rewriter.create<tosa::ReciprocalOp>(op->getLoc(), self.getType(), self);
+  // Converts AtenSqrtOp into pow(x, 0.5)
+  auto self = adaptor.getSelf();
+  auto selfTy = self.getType().dyn_cast<TensorType>();
+  if (!selfTy)
+    return rewriter.notifyMatchFailure(op,
+                                       "Only Tensor types supported in TOSA");
+
+  auto resultType = typeConverter->convertType(op.getType())
+                        .template cast<RankedTensorType>();
+  auto elementType = resultType.getElementType();
+
+  if (selfTy.getElementType().isa<mlir::IntegerType>()) {
+    self = rewriter.createOrFold<tosa::CastOp>(
+        op->getLoc(), RankedTensorType::get(resultType.getShape(), elementType),
+        self);
+  }
+
+  auto oneHalf =
```

Comment:
0.5 as integer is going to be zero right? So do the numbers match in the integer case?

Reply:
`torch` automatically infer the result type of sqrt done with `int` types as floating point. 
As far as I saw, what is needed to support `sqrt` in integers is the `tosa.cast` of the source to `resultTy` (implemented just above).